src/ert/ensemble_evaluator/evaluator.py

```diff
@@ -364,53 +375,35 @@ async def evaluator_server(self) -> None:
             else:
                 logger.debug("Got done signal. No dispatchers connected")
 
-            logger.debug("Waiting for batcher to finish...")
-            try:
-                await asyncio.wait_for(
-                    self._dispatcher.wait_until_finished(), timeout=20
-                )
-            except asyncio.TimeoutError:
-                logger.debug("Timed out waiting for batcher to finish")
-
             terminated_attrs: Dict[str, str] = {}
             terminated_data = None
             if self._result:
                 terminated_attrs["datacontenttype"] = "application/octet-stream"
                 terminated_data = cloudpickle.dumps(self._result)
 
             logger.debug("Sending termination-message to clients...")
+
             message = self._create_cloud_message(
                 EVTYPE_EE_TERMINATED,
                 data=terminated_data,
                 extra_attrs=terminated_attrs,
                 data_marshaller=cloudpickle.dumps,
             )
-            if self._clients:
-                # See note about return_exceptions=True above
-                await asyncio.gather(
-                    *[client.send(message) for client in self._clients],
-                    return_exceptions=True,
-                )
-
+            await self._messages_to_send.put(message)
+            await self._events.join()
+            await self._batch_processing_queue.join()
+            await self._messages_to_send.join()
         logger.debug("Async server exiting.")
 
-    def _run_server(self, loop: asyncio.AbstractEventLoop) -> None:
-        loop.run_until_complete(self.evaluator_server())
-        logger.debug("Server thread exiting.")
-
-    def start_running(self) -> None:
-        self._ws_thread.start()
-        self._ensemble.evaluate(self._config)
-
-    def _stop(self) -> None:
-        if not self._done.done():
-            self._done.set_result(None)
+    async def _stop(self) -> None:
+        self._server_done.set()
 
     def stop(self) -> None:
-        self._loop.call_soon_threadsafe(self._stop)
-        self._ws_thread.join()
+        assert self._loop
```

Comment:
this method is no longer called in the tests.

Reply:
It can be removed :+1: